app/routes/products_.$productId/route.tsx

```diff
@@ -182,3 +202,15 @@ export const links: LinksFunction = () => {
         },
     ];
 };
+
+function getInitialSelectedOptions(productOptions: products.ProductOption[] | undefined) {
+    const result: Record<string, string | null> = {};
+    for (const option of productOptions ?? []) {
+        if (option.name) {
+            const initialChoice = option?.choices?.length === 1 ? option.choices[0] : undefined;
+            result[option.name] = getChoiceValue(option, initialChoice) ?? null;
```

Comment:
maybe change `result` type to `string | undefined` to simply code here and in `selectedValue={selectedOptions[option.name!] ?? undefined}`

Reply:
fixed